### PR TITLE
Add react-popper for supercharged HKDropdowns

### PR DESCRIPTION
**Pull request by mattrothenberg**
Currently, `<HKDropdown>` looks beautiful on desktop-sized viewports. On smaller screens, though, the experience isn't as nice.

screen shot 2018-10-11 at 4 00 27 pm

screen shot 2018-10-11 at 3 59 16 pm

Since we're currently using CSS to position the Dropdown menu, we don't have tremendous flexibility (e.g, re-positioning the dropdown menu when the viewport size changes). As such, I've pulled in a 3rd-party library that wraps [Popper.js](https://popper.js.org/), one of the more-popular popover engines. This enables a whole bunch of goodies, most notably being able to dynamically update the position of the dropdown menu when screens get too small.

TODO/Wish List
- [ ] Expose more options than `Left` and `Right` for popover positioning. Popper has a whole slew of `placement` options. This would likely be a breaking change 😱
- [x] Figure out how to make `defaultProps` work... For some reason,. can't initialize the `align` prop to a default value

### CODE REVIEW

File: src/HKDropdown.tsx [hunk 1]
```diff
@@ -45,22 +46,37 @@ export default class HKDropdown extends React.Component<IDropdownProps, IDropdow
...
-        <div data-testid={`${name}-dropdown-content`} onClick={this.handleContentClick}>
-          {
-            showDropdown &&
-            (<ul className={classnames(alignDropdown, contentClassName)}>
-              {children}
-            </ul>)
-          }
-        </div>
-      </div>
+      <Manager>
```
Comment (idan) [inline src/HKDropdown.tsx]: Wondering if having multiple `Manager`s is bad in terms of resource usage versus sticking a single one somewhere in the DOM (kinda like react-malibu's spritesheet), but then it gets kinda ugly in terms of the UX. Have you tried this with multiple poppers in one page to make sure the multiple managers don't conflict?

File: src/HKDropdown.tsx [hunk 2]
```diff
@@ -45,22 +46,37 @@ export default class HKDropdown extends React.Component<IDropdownProps, IDropdow
...
+        <Reference>
+          {({ ref }) => (
+            <div className='relative dib' ref={ref}>
+              <HKButton onClick={this.handleDropdown} data-testid={`${name}-dropdown-button`} className={classnames({ ph1: !title }, className)} type={Type.Secondary} disabled={disabled}>
```
Comment (idan) [inline src/HKDropdown.tsx]: What's `data-testid`? this is new to me, I assume it's an affordance for testing?

File: src/declarations.d.ts [hunk 3]
```diff
@@ -8,3 +8,4 @@ declare module 'react-measure'
 declare module 'moment'
 declare module 'simple-react-modal'
 declare module 'classnames'
+declare module 'react-popper'
```
Comment (idan) [inline src/declarations.d.ts]: I think this is for modules that lack typescript definitions, but [react-popper has them](https://www.npmjs.com/package/react-popper#flow-and-typescript-types) so we ought to use them

File: stories/HKDropdown.stories.tsx [hunk 4]
```diff
@@ -52,7 +52,7 @@ const dropdownProps = [
 const stories = storiesOf(`HKDropdown`, module)
 dropdownProps.forEach((dropdown) => {
   stories.add(dropdown.storyName, () => (
-    <div className='flex justify-center'>
+    <div className=''>
```
Comment (idan) [inline stories/HKDropdown.stories.tsx]: Huh, why do we need the empty `className` here?

File: src/HKDropdown.tsx (hunk 1 above)
Comment (mattrothenberg) [reply src/HKDropdown.tsx]: I have not tried that – fantastic suggestion.

File: src/HKDropdown.tsx (hunk 2 above)
Comment (mattrothenberg) [reply src/HKDropdown.tsx]: This was actually here before this PR. I'll look into if/how we can get rid of this.

File: stories/HKDropdown.stories.tsx (hunk 4 above)
Comment (mattrothenberg) [reply stories/HKDropdown.stories.tsx]: We can either revert, or omit the `className` altogether. I believe this was here to horizontally center the stories themselves (which I'm not a huge fan of, personally()

File: src/declarations.d.ts (hunk 3 above)
Comment (mattrothenberg) [reply src/declarations.d.ts]: @idan I think I follow. How would we manually import `typings/react-popper.d.ts` instead?

**Pull request merged**